data/account.go

```diff
@@ -177,6 +178,39 @@ func (acc *Account) VerifyPassword(plain string) bool {
 	return err == nil
 }
 
+// UpdateEmail checks validity of a new e-mail address and updates the current account
+// with a valid new e-mail address.
+// The normal account update does not include the e-mail address for safety reasons.
+func (acc *Account) UpdateEmail(email string) error {
+	if !(len(email) > 2) || !strings.Contains(email, "@") {
+		return errors.New("Please use a valid e-mail address")
+	}
+	if len(email) > 512 {
+		return errors.New("Address too long, please shorten to 512 characters")
+	}
+	exists := &struct {
+		Email bool
+	}{}
+
+	const check = `SELECT (SELECT COUNT(*) FROM accounts WHERE email = $1) <> 0 AS email`
+	err := database.Get(exists, check, email)
+	if err != nil {
+		panic(err)
+	}
+	if exists.Email {
+		return errors.New("Please choose a different e-mail address")
```

Comment:
In `UpdateAccountEmail` this error is directly serialized to JSON. If it was a `ValidationError` the resulting error message could shown in the respective field in the form.